pkg/keystore/keystore.go

```diff
@@ -50,15 +53,20 @@ type Keystore struct {
 }
 
 // New returns a new instance of keystore.
-func New(keystoreID string, provider Provider) (*Keystore, error) {
+func New(keystoreID string, provider provider.Provider) (*Keystore, error) {
```

Comment:
Consider avoiding the repetition of names - "provider.Provider". Perhaps a different package name?

(sometimes the linter will complain about this)

Reply:
looks like it's quite common to name "main" type in the package after the name of the package itself (e.g. `crypto.Crypto`, `did.DID` from Aries framework, `template.Template`, `rand.Rand`, `image.Image` from Go standard library) but I do agree that here we may try to come up with a better name. I am planning some additional refactoring (as part of https://github.com/trustbloc/hub-kms/issues/17), so maybe something better will emerge as part of it (but can't promise).

Reply:
OK, that sounds reasonable to me. I think if an idea comes to mind then perhaps you could consider it, but it looks like you're right about this being common, so I think it's not much of a concern for me anymore